framework/include/timesteppers/Stepper.h

```diff
@@ -0,0 +1,442 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#ifndef STEPPERBLOCK_H
+#define STEPPERBLOCK_H
+
+#include <functional>
+
+#include "LinearInterpolation.h"
+#include "MooseUtils.h"
+#include "libmesh/numeric_vector.h"
+
+/// Holds all information used by Steppers to calculate dt via the "advance"
+/// function.
+class StepperInfo
+{
+public:
+  StepperInfo();
+  StepperInfo(const StepperInfo & si);
+  StepperInfo & operator=(const StepperInfo & si);
```

Comment:
You should deal with the move constructor and move assignment operator (i.e. either delete them or implement them).